scripts/west_commands/runners/nrfutil.py

```diff
@@ -82,6 +82,20 @@ def _insert_op(self, op):
         self._ops.append(op)
 
     def _exec_batch(self):
+        # Make sure nrfutil is recent enough
+        out = self._exec(['--version'])
+        version = tuple(int(x) for x in out[0]['data']['version'].split('.'))
+        if version < (2, 1, 1):
+            raise RuntimeError('nrfutil version 2.1.1 or later required, run `nrfutil upgrade`.')
+
+        # Make sure the device is in the expected family
+        out = self._exec(['device-info', '--serial-number', f'{self.dev_id}'])
```

Comment:
That is to executions of `nrfutil` only for pre-checks. This is gonna slow down the execution of the runner significantly

Reply:
Shouldn't it be the runner itself that cross-checks if the specified family matches the polled device, and error otherwise?

Reply:
The version check takes 0,228s. The family check takes 1,042s.

Reply:
The device family check is historically missing in nrfjprog and nrfutil. We could change the behavior of the --family option to stop if the family doesn't match, but that would be a breaking change.